lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -306,6 +320,8 @@ def main():
             func_kwargs.update({'Timeout': timeout})
         if memory_size and current_config['MemorySize'] != memory_size:
             func_kwargs.update({'MemorySize': memory_size})
+        if env_variables and current_config['Environment']['Variables'] != env_variables:
```

Comment:
This should probably be `if env_variables is not None` so that a user could pass an empty dictionary and wipe out the existing vars, but if they leave it blank the existing settings won't change.